### ISSUE TITLE
About model result evaluation

### ISSUE THREAD

**Issue by zhangshuaiqing** [dcaffo98/path-planning-cnn#9]
I once consulted you under the article you posted on Medium  [2D Path Planning With CNN](https://pub.towardsai.net/2d-path-planning-with-cnn-c7e3866b679d). Now, I have basically reproduced your program, but I have the following questions and would like to consult you again. The following are the results of the model result evaluation in picture form. In the picture, there seem to be obvious errors, such as directly passing through the white area, which is the obstacle area. Besides, it seems that the successfully planned path is significantly inferior to the D*lite algorithm. May I ask how I should adjust it to achieve more satisfactory results?
(It should be added that I downloaded the dataset through Kaggle [2d-path-planning-dataset](https://www.kaggle.com/datasets/dcaffo/2dpathplanningdataset) .)

The following is about passing through the white area:

![Image](https://github.com/user-attachments/assets/fd57d6e7-3099-4491-9a0d-57aa5eb414de)

![Image](https://github.com/user-attachments/assets/4e58a482-b641-4e45-a727-74b1dccf6cd7)

![Image](https://github.com/user-attachments/assets/8925b559-ba83-4cac-a4cd-41793d67c3b3)

The following is the result that is of poor quality:
![Image](https://github.com/user-attachments/assets/c73a40fc-20b2-4eae-b802-e5d38e2e5fcd)

![Image](https://github.com/user-attachments/assets/3cf66d58-5ce1-4e15-a398-6fe8c482a1c3)

**Comment by dcaffo98** [comment]
Hi @zhangshuaiqing, thank you for your interest in this work. 
Let me ask you a couple of questions:
1) for how long did you train the model? How many epochs?
2) what is the fraction of mazes with these errors concerning invalid paths?

**Comment by zhangshuaiqing** [comment]
Hi @dcaffo98 ,
1. It took me approximately 24 hours and 100 epochs to train the model.
2. In the summary.csv file, there are `51,103` records in total, among which `11,280` records have non-empty values for both `model_len` and `obstacle_hit`. The ratio of the two is `0.22073068117331662`. Are the ones where these two values are empty the invalid paths? If so, the proportion of invalid paths is `0.7792693188266834`.  Furthermore, the output of `evaluation.py` in terminal as followed:
```
Total test samples: 51103
Valid samples: 51103
```

Other parameter settings are as follows:
![Image](https://github.com/user-attachments/assets/b781e608-20eb-4747-9826-483de46196c7)
![Image](https://github.com/user-attachments/assets/ba25c338-3d04-4df9-9a8b-7e1dcaa1c66b)
![Image](https://github.com/user-attachments/assets/dd0ed588-cb5c-437f-863c-8907ffbb077f)

In addition, I'm not quite sure how to determine the invalid paths. I have pushed my results to that repository:
`model_results`: https://github.com/zhangshuaiqing/tmp/tree/master/path-planning-cnn/model_results
`model_results_evaluation`: https://github.com/zhangshuaiqing/tmp/tree/master/path-planning-cnn/model_results_evaluation
`summary.csv`: https://github.com/zhangshuaiqing/tmp/blob/master/path-planning-cnn/summary.csv

**Comment by dcaffo98** [comment]
>  the proportion of invalid paths is 0.7792693188266834

That's quite huge. Could it be overfitting? You said that you trained the model for 100 epochs, but originally I got the best checkpoint after 23 epochs.

**Comment by zhangshuaiqing** [comment]
The `validation loss` figure as followed. As shown in the figure, it can indeed be seen that the optimal checkpoint appears at the 23rd epoch.
![Image](https://github.com/user-attachments/assets/0779828d-4416-41ac-980e-83da0b7deef0).

**Comment by dcaffo98** [comment]
I do not know what's going on with your model,but I can provide you my [original checkpoint.](https://drive.google.com/file/d/13YPPRkzAMDDencW3JAEjLonLUi19lL1L/view?usp=drive_link)

**Comment by zhangshuaiqing** [comment]
I used the checkpoint you provided, and indeed, it performs better than the model I trained. The proportion of the  valid paths is 0.9501399135080132.

![Image](https://github.com/user-attachments/assets/ea4aee63-5836-408f-b7cc-582d126476c3)

![Image](https://github.com/user-attachments/assets/540de35e-e1e8-403f-9b10-06894e2b7df5)

![Image](https://github.com/user-attachments/assets/80408e9d-5f21-41a9-ab9b-09428ff43fef)

**Comment by dcaffo98** [comment]
> The proportion of the invalid paths is 0.9501399135080132

I guess (and hope) you meant that 0.95 is the proportion of the **_valid_** paths

**Comment by zhangshuaiqing** [comment]
I'm very sorry, And I meant that 0.95 is the proportion of the valid paths.
I will continue to check why such a situation occurred previously.
